OSDOCS-6962#adding custom IPv4 support (#86874)
openshift/openshift-docs: modules/installation-configuration-parameters.adoc

Set the `networking.machineNetwork` to match the CIDR that the preferred NIC resides in.
====

|networking:
  ovnKubernetesConfig:
    ipv4:
      internalJoinSubnet:
|
Configures the IPv4 join subnet that is used internally by `ovn-kubernetes`. This subnet must not overlap with any other subnet that {product-title} is using, including the node network. The size of the subnet must be larger than the number of nodes. You cannot change the value after installation.

> |An IP network block in CIDR notation. For example, `10.0.0.0/16`. The default value is `100.64.0.0/16`.

Reviewer: nit: I read `For example, `10.0.0.0/16`.` as boiler plate language we use with every cidr. So I'm fine leaving it in for consistency if that is the case. On the other hand, 10.0.0.0/16 would never be an appropriate value (10.0.0.0/16 would be for the machine network), so I think it is just as good, or a slight improvement to say:

```suggestion
|An IP network block in CIDR notation.  The default value is `100.64.0.0/16`.
```

Set the `networking.machineNetwork` to match the CIDR that the preferred NIC resides in.
====

|networking:
  ovnKubernetesConfig:
    ipv4:
      internalJoinSubnet:
|

> Configures the IPv4 subnet that is used internally by `ovn-kubernetes`.

Reviewer: ovn-kubernetes uses more than one subnet, so I would either say "configures a subnet" or, better:

Configures the IPv4 join subnet used internally...

Set the `networking.machineNetwork` to match the CIDR that the preferred NIC resides in.
====

|networking:
  ovnKubernetesConfig:
    ipv4:
      internalJoinSubnet:

> |When you configure `networkType` to `OVNKubernetes`, `OVNKubernetesConfig` provides the configuration for ovn-kubernetes as the default pod network.

Reviewer 2: `OVNKubernetes` is the only supported `networkType` since `openshift-sdn` was removed, so I don't think we need this line. @patrickdillon do you know if any other network plugins are in the works or may be added some day, or are we safe to assume `OVNKubernetes` is going to be the only network plugin for now?

Reviewer: Users are actually allowed to drop in other SDNs, such as calico.

> You can configure the IP settings for IPv4 connections. If you do not configure IPv4, the default configuration is used. For more information about the default configuration values, see the individual IPv4 fields.

Reviewer 2: I think this line is not necessary, since the `ipv4:` parameter is not really a parameter itself, but a category of parameters.

> `internalJoinSubnet` is a v4 subnet that is used internally by ovn-kubernetes in case the default one is already used by something else. `internalJoinSubnet` must not overlap with any other subnet that OpenShift is using or by the node network. The size of the subnet must be larger than the number of nodes. You cannot change the value after installation.

Reviewer 2: ```suggestion
Configures the IPv4 subnet that is used internally by `ovn-kubernetes`. This subnet must not overlap with any other subnet that {product-title} is using, including the node network. The size of the subnet must be larger than the number of nodes. You cannot change the value after installation.
```

> |The default value is `100.64.0.0/16`.
> 
> The subnet must be large enough to accommodate one IP per node in your cluster. The value must be in proper IPv4 CIDR format.

Reviewer 2: ```suggestion
|An IP network block in CIDR notation. For example, `10.0.0.0/16`. The default value is `100.64.0.0/16`.
```
